src/test/java/com/quaz/ticket/reservation/ReservationServiceImplTest.java

```diff
@@ -0,0 +1,155 @@
+        alreadyReservedScreeningSeat.setReservation(oldReservation);
+
+        final var newReservation = new Reservation();
+        newReservation.setReservedSeats(List.of(new ScreeningSeat(), alreadyReservedScreeningSeat));
+
+        //when
+        final var thrown = assertThrows(
+                IllegalArgumentException.class,
+                () -> reservationService.createReservation(newReservation)
+        );
+
+
+        //then
+        Assertions.assertEquals("Reservation should only be made for available seats", thrown.getMessage());
+    }
+
+    @Test
+    void givenReservationWithSeatFromAnotherScreeningWhenCreateReservationThenExceptionIsThrown() {
+        //given
+        final var screening = new Screening();
+        screening.setId(1L);
+        final var screeningSeat = new ScreeningSeat();
+        screeningSeat.setScreening(screening);
+
+        final var anotherScreening = new Screening();
+        anotherScreening.setId(2L);
+        final var anotherScreeningSeat = new ScreeningSeat();
+        anotherScreeningSeat.setScreening(anotherScreening);
+
+        final var reservation = new Reservation();
+        reservation.setReservedSeats(List.of(screeningSeat, anotherScreeningSeat));
+        reservation.setScreening(screening);
+
+        //when
+        final var thrown = assertThrows(
+                IllegalArgumentException.class,
+                () -> reservationService.createReservation(reservation)
+        );
+
+
+        //then
+        Assertions.assertEquals("Reservation should be made for one screening", thrown.getMessage());
+    }
+
+    @Test
+    void givenReservationTimeLessThan15MinutesBeforeScreeningWhenCreateReservationThenExceptionIsThrown() {
+        //given
+        final var screeningTime = TestConfig.fixedClock().instant().atOffset(UTC);
+        final var screening = new Screening();
+        screening.setScreeningTime(screeningTime);
+        final var screeningSeat = new ScreeningSeat();
+        screeningSeat.setScreening(screening);
+
+        final var reservation = new Reservation();
+        reservation.setReservedSeats(List.of(screeningSeat));
+        reservation.setScreening(screening);
+        reservation.setReservationTime(screeningTime.minus(Duration.ofMinutes(5)));
+
+        //when
+        final var thrown = assertThrows(
+                IllegalArgumentException.class,
+                () -> reservationService.createReservation(reservation)
+        );
+
+        //then
+        Assertions.assertEquals("Reservation should be made at least 15 minutes before screening time", thrown.getMessage());
+    }
+
+    @Test
+    void givenReservationThatLeavesEmptySpaceInRowWhenCreateReservationThenExceptionIsThrown() {
+        //given
+        final var screeningTime = TestConfig.fixedClock().instant().atOffset(UTC);
+        final var screening = new Screening();
+        screening.setScreeningTime(screeningTime);
+
+        final var seat1 = new Seat();
```

Comment:
Staram się wyznawać zasadę "pisz testy tak jak piszesz kod", a tutaj te testy są długie i mają dużo trywialnego kodu (pusty konstruktor i settery), który trochę zaciemnia to co się dzieje w teście. Spokojnie można z tego wydzielić prywatne metody (albo nawet jakiś TestUtilHelper), który tworzy te obiekty. Wtedy poniższy kod, który tworzy 3xSeat, a później 3x screening można zastąpić jednym wywołaniem:
```java
createScreeningSeatList(screening, 3);
```
Test sporo zyskuje na czytelności, nie sądzisz? :)

Reply:
Prawda, dzięki za uwagę. Zapamiętam tę zasadę :D